### PR TITLE
[gui-tests] Refactor hooks and config options

### PR DESCRIPTION
**Pull request by saw-jan**
In this PR, I have refactored the tests code related to hooks and configdir. It is better to use single source for paths. I have moved hooks from the steps to the bdd_hooks.

### CODE REVIEW

**Comment by ownclouders** [comment]
Results for <strong>GUI-tests</strong> https://drone.owncloud.com/owncloud/client/13015/6/1

<details>


</details>

File: test/gui/shared/scripts/bdd_hooks.py [hunk 1]
```diff
@@ -76,11 +78,11 @@ def hook(context):
...
     try:
-        cfg.read('../config.ini')
-        for key, value in context.userData.items():
-            if value == '':
-                context.userData[key] = cfg.get('DEFAULT', CONFIG_ENV_MAP[key])
-    except Exception as err:
+        if cfg.read('../config.ini'):
+            for key, value in context.userData.items():
+                if value == '':
+                    context.userData[key] = cfg.get('DEFAULT', CONFIG_ENV_MAP[key])
```
Comment (saw-jan) [inline test/gui/shared/scripts/bdd_hooks.py:84]: Don't try to check for values if `cfg.read()` returns empty.
This prevents logs like `No option 'max_sync_timeout' in section: 'DEFAULT'`

**Comment by saw-jan** [comment]
> Results for **GUI-tests** https://drone.owncloud.com/owncloud/client/13026/6/1
> 
> boom The GUI tests failed.
> 
> GUI Logs: https://cache.owncloud.com/public/owncloud/client/13026/guiReportUpload/index.html
> 
> Server Logs: https://cache.owncloud.com/public/owncloud/client/13026/guiReportUpload/serverlog.log
> 
> Screenshots:
> 
> * [Delete_a_file.png](https://cache.owncloud.com/public/owncloud/client/13026/guiReportUpload/screenshots/Delete_a_file.png)

Intentional failure. Working fine

**Review by amrita-shrestha — COMMENTED** [review]
.

File: test/gui/shared/scripts/bdd_hooks.py [hunk 2]
```diff
@@ -125,11 +157,7 @@ def hook(context):
     global previousErrorResultCount
```
Comment (amrita-shrestha) [inline test/gui/shared/scripts/bdd_hooks.py:157]: why use the global variable twice? can't we use once

Comment (saw-jan) [reply test/gui/shared/scripts/bdd_hooks.py:157]: no twice!
functions have their own scopes

**Review by amrita-shrestha — APPROVED** [review]
lgtm

**Pull request merged**